sumpy/expansion/local.py

```diff
@@ -189,79 +312,44 @@ def translate_from(self, src_expansion, src_coeff_exprs, src_rscale,
-                src_expansion.expansion_terms_wrangler.copy(
-                        order=self.order + src_expansion.order, max_mi=tuple(max_mi))
-            tgtplusderiv_coeff_ids = \
-                tgtplusderiv_exp_terms_wrangler.get_coefficient_identifiers()
-            tgtplusderiv_full_coeff_ids = \
-                tgtplusderiv_exp_terms_wrangler.get_full_coefficient_identifiers()
+            # Calculate the first row of the upper triangular Toeplitz matrix
+            toeplitz_first_row = [0] * len(toeplitz_matrix_coeffs)
+            for coeff, term in zip(
+                    src_coeff_exprs,
+                    src_expansion.get_coefficient_identifiers()):
+                toeplitz_first_row[toeplitz_matrix_ident_to_index[term]] = \
+                        add_to_sac(sac, coeff)
 
-            tgtplusderiv_ident_to_index = {ident: i for i, ident in
-                                enumerate(tgtplusderiv_full_coeff_ids)}
+            # Do the matvec
+            output = matvec_toeplitz_upper_triangular(toeplitz_first_row,
+                                derivatives_full)
 
+            # Filter out the dummy rows and scale them for target
             result = []
-            for lexp_mi in self.get_coefficient_identifiers():
-                lexp_mi_terms = []
-
-                # Embed the source coefficients in the coefficient array
-                # for the (tgt order)+(src order) wrangler, offset by lexp_mi.
-                embedded_coeffs = [0] * len(tgtplusderiv_full_coeff_ids)
-                for coeff, term in zip(
-                        src_coeff_exprs,
-                        src_expansion.get_coefficient_identifiers()):
-                    embedded_coeffs[
-                        tgtplusderiv_ident_to_index[add_mi(lexp_mi, term)]] \
-                                    = coeff
-
-                # Compress the embedded coefficient set
-                stored_coeffs = tgtplusderiv_exp_terms_wrangler \
-                        .get_stored_mpole_coefficients_from_full(
-                                embedded_coeffs, src_rscale, sac=sac)
-
-                # Sum the embedded coefficient set
-                for tgtplusderiv_coeff_id, coeff in zip(tgtplusderiv_coeff_ids,
-                                                        stored_coeffs):
-                    if coeff == 0:
-                        continue
-                    kernel_deriv = taker.diff(tgtplusderiv_coeff_id)
-                    rscale_ratio = tgt_rscale / src_rscale
-                    lexp_mi_terms.append(
-                            coeff * kernel_deriv * rscale_ratio**sum(lexp_mi))
-                result.append(sym.Add(*lexp_mi_terms))
+            rscale_ratio = add_to_sac(sac, tgt_rscale/src_rscale)
+            for term in self.get_coefficient_identifiers():
+                index = toeplitz_matrix_ident_to_index[term]
+                result.append(output[index]*rscale_ratio**sum(term))
```

Comment:
Use vim-style `{{{` / `}}}` range comments here?

Reply:
Fixed in 8deb5d9